### PR TITLE
update uct_def.h

### PR DESCRIPTION
**Pull request by shengyang-3390**
## What
update UCT_COMPONENT_NAME_MAX     

## Why ?
device names may be larger than 16 for ROCE, for example ib/hnsenp189s0f2

## How ?
update UCT_COMPONENT_NAME_MAX = 32

### CODE REVIEW

File: src/uct/api/uct_def.h [hunk 1]
```diff
@@ -15,9 +15,9 @@
 #include <sys/types.h>
 
 
-#define UCT_COMPONENT_NAME_MAX     16
```
Comment (yosefe) [inline src/uct/api/uct_def.h]: component name is defined in compile time and not runtime

File: src/uct/api/uct_def.h [hunk 2]
```diff
@@ -15,9 +15,9 @@
...
+#define UCT_COMPONENT_NAME_MAX     32
 #define UCT_TL_NAME_MAX            10
-#define UCT_MD_NAME_MAX            16
```
Comment (yosefe) [inline src/uct/api/uct_def.h]: memory domain name does not include "ib/" prefix anymore.
is this still needed?

Comment (shengyang-3390) [reply src/uct/api/uct_def.h]: i think that the change of UCT_MD_NAME_MAX may be needed.
when I use ucx_info -d, it shows that 
image

Comment (yosefe) [reply src/uct/api/uct_def.h]: @shenyang-3390 are you using latest master branch?

Comment (shengyang-3390) [reply src/uct/api/uct_def.h]: the version I am using is 1.6.0
the latest master is ok. could you update it for 1.6?

Comment (yosefe) [reply src/uct/api/uct_def.h]: i'm afraid that for v1.6 it's too big change which includes also API change
I'd suggest to use https://github.com/openucx/ucx/releases/tag/v1.7.0-rc1 which also has this, and going to be released soon

Comment (shengyang-3390) [reply src/uct/api/uct_def.h]: ok ,3ks

**Pull request closed without merging**